src/stream.rs

```diff
@@ -2,16 +2,52 @@ use std::pin::Pin;
 
 use tokio::io::{AsyncRead, AsyncWrite};
 
-// TODO: Reconsider this. Do we really need Stream + AnyStream? What is the smallest API that we need to expose?
+trait AsyncReadWrite: AsyncRead + AsyncWrite + Send {
+    fn read(self: Pin<&Self>) -> Pin<&(dyn AsyncRead + Send)>;
+    fn read_mut(self: Pin<&mut Self>) -> Pin<&mut (dyn AsyncRead + Send)>;
+    fn write(self: Pin<&Self>) -> Pin<&(dyn AsyncWrite + Send)>;
+    fn write_mut(self: Pin<&mut Self>) -> Pin<&mut (dyn AsyncWrite + Send)>;
```

Comment:
Question: Should we name these `reader`, `reader_mut`, `writer`, `writer_mut`, instead?